test/dynamo/test_functions.py

```diff
@@ -2010,6 +2013,18 @@ def fn(x):
                 x = torch.randn(10)
                 self.assertEqual(opt_fn(x), fn(x))
 
+    def test_compare_tensor_and_constant(self):
+        for op in comparison_ops:
+            with self.subTest(op=op):
+
+                def fn(x):
+                    return op(x, -10)
+
+                opt_fn = torch.compile(fullgraph=True)(fn)
```

Comment:
Use torch._dynamo.optimize or torch.compile(backend="eager"). This is test/dynamo -- we don't generally run things through all of PT2 stack in tests in this directory.